src/test/java/org/junitpioneer/internal/TestExtensionContext.java

```diff
@@ -35,12 +35,12 @@ public TestExtensionContext(Class<?> testClass, Method testMethod) {
 
 	@Override
 	public Optional<Class<?>> getTestClass() {
-		return Optional.of(testClass);
+		return Optional.ofNullable(testClass);
 	}
 
 	@Override
 	public Optional<Method> getTestMethod() {
-		return Optional.of(testMethod);
+		return Optional.ofNullable(testMethod);
```

Comment:
Can you explain why these changes are needed?

Reply:
@nipafx It's so that I can make a `TestExtensionContext` that has no test method.

This is to check an (admittedly extremely unlikely) edge case in `ResourceManagerExtension` where all hell has broken loose and:

1. `ResourceManagerExtension::supportsParameter` thinks a parameter has `@New`, but `::resolveParameter` disagrees.
2. And JUnit 5 itself gives `::resolveParameter` an `ExtensionContext` with an absent `getTestMethod()`.

My thinking was to ensure a helpful exception is thrown even in this case.

But I can see now that the change to `TestExtensionContext::getTestClass` is not needed. And regardless, this generally feels convoluted.

So I'm happy to revert this hunk and avoid testing this edge case if you'd like?

(For context, here's the [edge case](https://github.com/junit-pioneer/junit-pioneer/pull/491/files#diff-23a9e64b2f36fe2125ffc4a2aa13641e97c9c213cd3e0cffbc0a1b30aaa9e753R41) I talked about.)